zim/tokenparser.py

```diff
@@ -21,9 +21,16 @@ class EndOfTokenListError(AssertionError):
 	pass
 
 
-def collect_untill_end_token(token_iter, tag):
+def collect_until_end_token(token_iter, tag):
```

Comment:
Good catch on the typo in the function name.

Would prefer to revers this function to the implementation without a generator. Function is small and low level so called often in parsing so rather keep it a single function call. In this better to have a bit of redundancy in functions